### PR TITLE
DEVDOCS-4120: document GraphQL error messages

### PR DESCRIPTION
**Pull request by slsriehl**
# [DEVDOCS-4120](https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4120)

## What changed?
* revise api-status-codes to accommodate graphQL error messages
* add link to main article

### CODE REVIEW

**Comment by bc-andreadao** [comment]
See [PR from storefront repository ](https://github.com/bigcommerce/storefront/pull/1590)

File: docs/api-docs/getting-started/api-status-codes.md [hunk 1]
```diff
@@ -43,29 +45,43 @@ The BigCommerce API responds to each request with an HTTP status code that depen
...
+|**403**| App lacks required OAuth scopes, a store-owner account changed, operations resulting from API request exceed a platform limit, or URL requested is incorrect.|Double-check OAuth Scopes in **control panel** > **API Accounts** or in **Developer Portal** > **My Apps**.|
+|||Check the URL. Are the endpoint and store hash correct?|
+|||Ensure [platform limits](https://support.bigcommerce.com/s/article/Platform-Limits#product-catalog-limits) have not been reached.|
+|**415**| Request headers specify an unsupported `content-type` (or header is missing).|Double-check `content-type` request header.|
+|**500**|Expensive API calls or an internal server error in BigCommerce.|Re-attempt the request three to five times, with increasing delays of at least a minute between attempts.|
+|||Try reducing the number of objects being requested. You can request fewer objects in the v2 API, by using `?limit={count}`. In  `v2` and `v3` API, fewer objects can be requested by excluding certain fields or only requesting certain fields.|
+||| Check the BigCommerce [Status Page](https://status.bigcommerce.com/).|
+
+## GraphQL API HTTP status codes
+
```
Comment (slsriehl) [inline docs/api-docs/getting-started/api-status-codes.md:76]:
```suggestion

All GraphQL errors return a 401 HTTP status code.

```

File: docs/api-docs/getting-started/api-status-codes.md [hunk 2]
```diff
@@ -43,29 +45,43 @@ The BigCommerce API responds to each request with an HTTP status code that depen
...
+| Code | Text | Purpose |
+|:----:|:-----|:--------|
+| **401** | **Unauthorized** | This response is sent when your client failed to provide credentials or its credentials were invalid. |
+| | "GraphQL credentials were missing. No token was sent." | No token sent (missing token). |
+| | "GraphQL invalid credentials. String is not a JWT" | Invalid JWT sent (token was not a JWT). |
+| | "GraphQL invalid credentials. JWT signature is invalid" | One of the JWT claims failed (token was not generated by BC). |
+| | "GraphQL invalid credentials. JWT is expired" | Token has expired. |
+| | "GraphQL invalid credentials. JWT channel id doesn't match channel id of the URL" | Channel ID mismatch (channel ID in the token does not match channel ID of the site/URL). |
+| | "Unknown JWT token" | JWT was revoked. |
+| | "GraphQL invalid credentials. JWT has no valid Json" or "GraphQL invalid credentials. JWT has Json of an unknown format" | Invalid JWT sent. Token has no valid JSON. |
```
Comment (slsriehl) [inline docs/api-docs/getting-started/api-status-codes.md:86]: break this line up into 2 lines.

File: docs/api-docs/storefront/graphql/graphql-api-overview.md [hunk 3]
```diff
@@ -596,6 +596,10 @@ Use normal GraphQL Storefront API tokens. According to the [Principle of least p
 
 Use a Customer Impersonation token and store it securely on your server like other secrets. When you need to run requests in the context of a particular customer (for example, if they've logged in to your application), send their BigCommerce Customer ID along with the request as the `X-Bc-Customer-Id` header.
 
+### I want a list of GraphQL error messages
+
+For a list of GraphQL error messages, see [API Status Codes](/api-docs/getting-started/api-status-codes).
```
Comment (slsriehl) [inline docs/api-docs/storefront/graphql/graphql-api-overview.md:601]:
```suggestion
For a list of GraphQL error messages, see [API Status Codes](/api-docs/getting-started/api-status-codes#graphql-api-http-status-codes).

```

File: docs/api-docs/getting-started/api-status-codes.md (hunk 2 above)
Comment (bc-andreadao) [reply docs/api-docs/getting-started/api-status-codes.md]: done

**Comment by slsriehl** [comment]
I opened the PR, but assigned it. Approved.

**Pull request merged**